core/tests/protractor_desktop/adminMiscTab.js

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end tests for admin page functionality.
+ */
+
+var general = require('../protractor_utils/general.js');
+var users = require('../protractor_utils/users.js');
+var AdminPage = require('../protractor_utils/AdminPage.js');
+var TopicsAndSkillsDashboardPage =
+  require('../protractor_utils/TopicsAndSkillsDashboardPage.js');
+var TopicEditorPage = require('../protractor_utils/TopicEditorPage.js');
+
+describe('Admin misc test tab', function() {
+  var adminPage = null;
+  var topicsAndSkillsDashboardPage = null;
+  var topicEditorPage = null;
+  var topicId = null;
+  var allowedErrors = ['Failed to load resource', 'Object', 'Entity',
+    '500', 'encode'];
+
+  beforeAll(async function() {
+    adminPage = new AdminPage.AdminPage();
+    topicsAndSkillsDashboardPage = (
+      new TopicsAndSkillsDashboardPage.TopicsAndSkillsDashboardPage());
+    topicEditorPage = new TopicEditorPage.TopicEditorPage();
+
+    await users.createAndLoginAdminUser(
+      'miscTabTester@miscTab.com', 'miscTabTester');
+
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.createTopic('miscTabTopic',
+      'A topic to test the misc tab', false);
+    await browser.getCurrentUrl().then((url) => {
+      var startIndex = url.split('/', 4).join('/').length + 1;
+      topicId = url.substring(startIndex, url.length - 2);
+    });
+    await topicEditorPage.publishTopic();
+  });
+
+  it('should upload similarity file', async function() {
+    await adminPage.get();
+    await adminPage.getMiscTab();
+    await adminPage.uploadTopicSimilarities(
+      '../data/sample_topic_similarities.csv');
+    await adminPage.expectSimilaritiesToBeUploaded();
+    await adminPage.uploadTopicSimilarities('../data/cafe.mp3');
+    await adminPage.expectUploadError();
+  });
+
+  it('should download similarity files', async function() {
+    await adminPage.downloadSimilarityFile();
+    await adminPage.expectFileToBeDownloaded();
+  });
```

Comment:
Could you combine these two? This would bring 2 advantages:
1. The test would be independent, while right now the download test has to run after the upload test
2. We could check that the downloaded file matches the uploaded one. This would catch errors where, e.g. the file appears to upload but gets mangled

Reply:
1) I've combined them

2) I tried for a couple of hours to make this work, but I couldn't find a way to access the downloaded file. I looked extensively for a solution but to no avail. Do you have any ideas/anything you would like me to try?

Reply:
Here are some solutions I found online:
* https://stackoverflow.com/questions/21935696/protractor-e2e-test-case-for-downloading-pdf-file
* https://sqa.stackexchange.com/questions/24175/how-to-check-if-a-file-with-random-changing-name-is-downloaded-using-protractor
https://sqa.stackexchange.com/questions/24013/how-to-check-if-a-file-is-downloaded-using-protractor-selenium

I see you're importing `fs`, so maybe you already tried these. If they help you figure out a way that works, then great. If not, then don't worry about comparing the files

Reply:
I tried those solutions, but I wasn't able to find a solution that worked for my test :/ 

Perhaps this is something we can incorporate in a future PR?

Reply:
If you want to work on this in a future PR, go for it, but I also don't think we need to worry about it. It would be nice to compare the files, but it's not too important